packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js

```diff
@@ -107,21 +102,19 @@ export default class VotingPage extends Component<Props> {
         </VerticallyCenteredLayout>
       );
     }
+    const { currentFund } = this.generated.stores.substores.ada.votingStore.catalystRoundInfo
 
     // keep enabled on the testnet
     if (!environment.isTest()) {
-      const isLate = new Date() >= roundInfo.endDate;
-      const isEarly = new Date() <= roundInfo.startDate;
+       const isLate = new Date() >= new Date(Date.parse(currentFund.registrationEnd))
+       const isEarly = new Date() <= new Date(Date.parse(currentFund.registrationStart))
+       const roundNumber = isLate ? currentFund.id : currentFund.id + 1
       if (
         selected.getParent().getNetworkInfo().NetworkId === networks.CardanoMainnet.NetworkId &&
         (isEarly || isLate)
       ) {
         return (
-          <RegistrationOver roundNumber={
-              isLate
-                ? roundInfo.nextRound
-                : roundInfo.nextRound - 1
-            }
+          <RegistrationOver roundNumber={roundNumber}
```

Comment:
I think we should do this:
```
if `currentFund` is available {
 if it is before `registrationStart` {
  show 'round {currentFund.id} registration starts at {registrationStart}'
 }
 if it is after `registrationEnd` {
  if it is before `votingStart` {
    show "Registration has ended. Voting starts at {votingStart}"
  }
  if it is between `votingStart` and `votingEnd` {
   show "Registration has ended.  Voting ends at  {votingEnd}"
  }
  if it is after `votingEnd` {
    show "Round {currentFund.id} ended.
    if `nextFund` is available {
      show "Round {nextFund.id} starts at {nextFund.registrationStart}"
    }
  }
 }
} else { // `currentFund` is not available
 if nextFund is available {
   show "Round {nextFund.id} starts at {nextFund.registrationStart}"
 } else {
   show a generic "registration is over" page with no specific info
 }
}
``` 
cc @vsubhuman

Reply:
This is what I am talking about
@vsubhuman  Can you please confirm this so I can start